src/lib/konnectors.js

```diff
@@ -32,11 +32,22 @@ export function findBySlug (cozy, slug) {
     .then(list => list.length ? list[0] : null)
 }
 
-export function install (cozy, slug, source, timeout = 120000) {
-  if (!slug) throw new Error('Missing `slug` paramater for konnector')
-  if (!source) throw new Error('Missing `source` parameter for konnector')
+export function install (cozy, konnector, timeout = 120000) {
+  ['slug', 'source'].forEach(property => {
+    if (!konnector[property]) throw new Error(`Missing '${property}' property in konnector`)
+  })
+
+  const { slug, source } = konnector
 
-  return cozy.fetchJSON('POST', `/konnectors/${slug}?Source=${encodeURIComponent(source)}`)
+  return findBySlug(cozy, slug)
+    .catch(error => {
+      if (error.status !== '404') throw error
+      return null
```

Comment:
Actually here, we are testing if the konnector is already installed. It is not the case if the `findBySlug` returns a 404 error. So we handle this properly and we don't need to warn the user.